extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/HttpServerInitializer.java

```diff
@@ -0,0 +1,160 @@
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CountDownLatch;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.Timing;
+import io.quarkus.vertx.web.Route;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+
+@Singleton
+public class HttpServerInitializer {
+
+    private static final Logger LOGGER = Logger.getLogger(HttpServerInitializer.class.getName());
+
+    private volatile Router router;
+
+    private volatile HttpServer httpServer;
+
+    @Inject
+    Vertx vertx;
+
+    @Inject
+    Event<Router> routerEvent;
+
+    void initialize(VertxHttpConfiguration vertxHttpConfiguration, Map<String, List<Route>> routeHandlers,
+            LaunchMode launchMode) {
+        Router router = Router.router(vertx);
+        router.route().handler(BodyHandler.create());
+        for (Entry<String, List<Route>> entry : routeHandlers.entrySet()) {
+            Handler<RoutingContext> handler = createHandler(entry.getKey());
+            for (Route route : entry.getValue()) {
+                addRoute(router, handler, route);
+            }
+        }
+        // Make it also possible to register the route handlers programatically
+        routerEvent.fire(router);
+        // Start the server
+        CountDownLatch latch = new CountDownLatch(1);
+        HttpServerOptions httpServerOptions = createHttpServerOptions(vertxHttpConfiguration, launchMode);
+        this.router = router;
+        httpServer = vertx.createHttpServer(httpServerOptions).requestHandler(router)
+                .listen(ar -> {
+                    if (ar.succeeded()) {
+                        // TODO log proper message
+                        Timing.setHttpServer(String.format(
+                                "Listening on: http://%s:%s", httpServerOptions.getHost(), httpServerOptions.getPort()));
+                        latch.countDown();
+                    }
+                });
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException("Unable to start the HTTP server", e);
+        }
+    }
+
+    // Note that we need a client proxy because if a bean also @Observes Router a null value would be injected 
+    @ApplicationScoped
+    @Produces
+    Router produceRouter() {
+        return router;
```

Comment:
We should produce the RX and Axle router also.

Reply:
This is something I wanted to ask you @cescoffier. So I suppose that RX and Axle versions built on top of a regular router and we should follow the pattern from [VertxProducer](https://github.com/quarkusio/quarkus/blob/master/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxProducer.java#L43). But what's the advantage of the rx version... let's say for `Router`?

Reply:
I agree that for the Router itself the RX API does not bring much. So, I won't mind not exposing it.

Reply:
Ok, we do produce the RX one now so let's keep it ;-)